src/prover.rs

```diff
@@ -144,6 +144,32 @@ where
     })
 }
 
+/// Given a Groth16 proof, returns a proof of the same statement. The proof that is returned is
+/// computationally unlinkable to the given proof, assuming the RNG is cryptographically secure
+pub fn rerandomize_proof<E, R>(rng: &mut R, vk: &VerifyingKey<E>, proof: &Proof<E>) -> Proof<E>
+where
+    E: PairingEngine,
+    R: Rng,
+{
+    // These are our rerandomization factors
+    let (r, s) = (E::Fr::rand(rng), E::Fr::rand(rng));
```

Comment:
Note unrelated to this PR: We should add a method to `ark-std` for sampling non-zero elements